docs/content/ingestion/native_tasks.md

```diff
@@ -0,0 +1,523 @@
+
+#### TuningConfig
+
+The tuningConfig is optional and default parameters will be used if no tuningConfig is specified. See below for more details.
+
+|property|description|default|required?|
+|--------|-----------|-------|---------|
+|type|The task type, this should always be `index_single_phase_parallel`.|none|yes|
+|targetPartitionSize|Used in sharding. Determines how many rows are in each segment.|5000000|no|
+|maxRowsInMemory|Used in determining when intermediate persists to disk should occur.|75000|no|
+|maxTotalRows|Total number of rows in segments waiting for being published. Used in determining when intermediate publish should occur.|150000|no|
+|numShards|Directly specify the number of shards to create. If this is specified and 'intervals' is specified in the granularitySpec, the index task can skip the determine intervals/partitions pass through the data. numShards cannot be specified if targetPartitionSize is set.|null|no|
+|indexSpec|defines segment storage format options to be used at indexing time, see [IndexSpec](#indexspec)|null|no|
+|maxPendingPersists|Maximum number of persists that can be pending but not started. If this limit would be exceeded by a new intermediate persist, ingestion will block until the currently-running persist finishes. Maximum heap memory usage for indexing scales with maxRowsInMemory * (2 + maxPendingPersists).|0 (meaning one persist can be running concurrently with ingestion, and none can be queued up)|no|
+|forceExtendableShardSpecs|Forces use of extendable shardSpecs. Experimental feature intended for use with the [Kafka indexing service extension](../development/extensions-core/kafka-ingestion.html).|false|no|
+|reportParseExceptions|If true, exceptions encountered during parsing will be thrown and will halt ingestion; if false, unparseable rows and fields will be skipped.|false|no|
+|pushTimeout|Milliseconds to wait for pushing segments. It must be >= 0, where 0 means to wait forever.|0|no|
+|segmentWriteOutMediumFactory|Segment write-out medium to use when creating segments. See [Indexing Service Configuration](../configuration/indexing-service.html) page, "SegmentWriteOutMediumFactory" section for explanation and available options.|Not specified, the value from `druid.peon.defaultSegmentWriteOutMediumFactory` is used|no|
+|maxNumBatchTasks|Maximum number of tasks which can be run at the same time.|Integer.MAX_VALUE|no|
+|maxRetry|Maximum number of retries on task failures.|3|no|
+|taskStatusCheckPeriodMs|Polling period in milleseconds to check running task statuses.|1000|no|
+|chatHandlerTimeout|Timeout for reporting the pushed segments in worker tasks.|PT10S|no|
+|chatHandlerNumRetries|Retries for reporting the pushed segments in worker tasks.|5|no|
+
+#### HTTP Endpoints
+
+The supervisor task provides some HTTP endpoints to get running status.
+
+* `http://{PEON_IP}:{PEON_PORT}/druid/worker/v1/chat/{SUPERVISOR_TASK_ID}/isRunningInParallel`
```

Comment:
nit: I think this would read a little easier if these are h5 elements ("#####") instead of list, making the description  be under the heading

Reply:
Changed.

Reply:
Oof, my bad, I guess you were being consistent with this style, https://github.com/druid-io/druid/blob/d981a2aa02aedfdf18e0bed5848e704bd9597563/docs/content/design/middlemanager.md#get, would probably be better to stick to that.

Reply:
Good point. Reverted.